service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseConverter.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.hmcts.reform.fpl.utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+
+import java.util.Map;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CaseConverter {
+    private final ObjectMapper mapper;
+
+    @SuppressWarnings("unchecked")
+    public Map<String, Object> convertToMap(CaseData caseData) {
+        return mapper.convertValue(caseData, Map.class);
```

Comment:
I would suggest changing `Map.class` to `new TypeReference<>(){}` which would then allow the removal of the suppress warnings